src/plugins/formulas/__tests__/publicAPI.spec.js

```diff
@@ -72,15 +72,15 @@ describe('Formulas public API', () => {
         }
       });
 
-      setDataAtCell(2, 0, '{=TRANSPOSE(A1:B2)}');
+      setDataAtCell(2, 0, '=TRANSPOSE(A1:B2)');
 
       const formulas = getPlugin('formulas');
 
       expect(formulas.getCellType(0, 0)).toBe('VALUE');
       expect(formulas.getCellType(0, 1)).toBe('VALUE');
       expect(formulas.getCellType(1, 0)).toBe('VALUE');
-      expect(formulas.getCellType(2, 0)).toBe('MATRIX');
-      expect(formulas.getCellType(3, 0)).toBe('MATRIX');
+      expect(formulas.getCellType(2, 0)).toBe('ARRAYFORMULA');
+      expect(formulas.getCellType(3, 0)).toBe('VALUE');
```

Comment:
Isn't this going to be ARRAY type? The default configuration has `useArrayArithmetic` disabled. But then, we need to either change `=TRANSPOSE(A1:B2)` to `=ARRAYFORMULA(A2:A5*B2:B5)` that will result in an array or add another test where `useArrayArithmetic` is enabled. ATM we have just one edge case

```suggestion
      expect(formulas.getCellType(3, 0)).toBe('ARRAY');
```

Reply:
Arrays don't seem to work that well at the moment (this spec produces a `SPILL` error), I only made this test pass by changing the spec definition. Matrices were broken in the past as well.

Reply:
This seems to have been caused by array-based functions' results not being correctly transferred to handsontable, I think it should work fine after https://github.com/handsontable/handsontable/pull/8669/commits/c52a4f85f0966f7c2eeed8e267a374aff3480695. (also, I modified the test case to use both `TRANSPOSE` and `ARRAYFORMULA`)